Fix memory leak in jint.c

xexyl opened the pull request:
free() was called if the pointer was == NULL rather than != NULL.

lcn2 commented:
Oops 🫣, sorry (tm Canada🇨🇦) and thanks @xexyl for the catch!

xexyl commented:
> Oops 🫣, sorry (tm Canada🇨🇦) and thanks @xexyl for the catch!

It happens to everyone and you're quite welcome!

Hopefully more from me tomorrow but we'll see. Have a nice day!

lcn2 commented:
> > Oops 🫣, sorry (tm Canada🇨🇦) and thanks @xexyl for the catch!
> 
> 
> 
> It happens to everyone and you're quite welcome!
> 
> 
> 
> Hopefully more from me tomorrow but we'll see. Have a nice day!

We wish more people would appreciate when someone else finds a flaw in code. Avoiding attaching "ego" to code is critical to developing and maintaining good code.

xexyl commented:
> > > Oops 🫣, sorry (tm Canada🇨🇦) and thanks @xexyl for the catch!
> > 
> > 
> > It happens to everyone and you're quite welcome!
> > Hopefully more from me tomorrow but we'll see. Have a nice day!
> 
> We wish more people would appreciate when someone else finds a flaw in code. Avoiding attaching "ego" to code is critical to developing and maintaining good code.

Well said and you bring up very real tragic phenomena. 

You have users who are rude to developers even of software that they don’t pay any amount for. And you might have some sympathy for some of the developers but not all because some developers are just as insolent. More than once on Red Hat bugzilla I have tried to get developers and users to be more respectful towards each other. It doesn’t really work out. 

I totally understand frustration with some developers. systemd comes to mind as the biggest offender. But there’s no need to be rude to them and it doesn’t make them want to help. 

For the MUD I was heavily involved in I got both: many (most) players loved all the things I added and taking care of bugs etc. But there were some who were nasty even about simple innocent mistakes. My approach with them was to always put their requests at the bottom of the queue and if anything new came up (which might be simply because I thought of something new - and I wasn’t against coming up with something strictly so I could do something else) I would push their request further behind. Unless of course it was a bug that had to be fixed. There was no reason or excuse for them to be rude and if they couldn’t treat me with respect I wouldn’t treat them at all. I just ignored them. It was after all my free time and doing it for fun. 

But then you have another problem: terrible bug reports. Some simply are just ignorance of how to make a good report.  But others are deliberately unhelpful. Things like ‘it doesn’t work’, ‘I want this or that’, ‘it crashes’ or whatever else. Maybe in some cases those types of reports are ignorance but when they do it repeatedly it’s simply not bothering to help. Which is something I have yet to figure out. They want help but they don’t give any information at all! How can someone help you with a problem if all you say is ‘it doesn’t work’?

And then there’s the lack of please and thank you! I actually add ‘please’ to all my requests to **Siri**! 

If everyone treated everyone else nicely the world would be a much better place. Unfortunately though this doesn’t seem to be something mankind is capable of and instead what it is capable of is being nasty and it’s rather in human nature to destroy ourselves and everything else. 

But I really appreciate your words! I wish more people understood them though. Nobody is perfect. We all make mistakes but really the only TRUE mistake is the mistake you refuse to acknowledge is a mistake and make better. 

Okay yes some mistakes you can’t repair but you can at least try and be better and apologise when you’re in the wrong (a genuine apology - not a fake one which sadly many people do). Some things you can’t take back but you can at least try and make sure you don’t do the same thing again. Unless of course you ended up getting a Darwin Award (or is it Darwin’s? It’s been a long time since I have seen it) in the process I guess!

xexyl commented:
Anyway I am going to try and get my head down now. Have a great rest of your day! I don’t know what I will end up doing tomorrow but hopefully I will get a little reading in if nothing else. 

But it might be that I take another day or two off. We shall see. 

Good night!

lcn2 commented:
Interesting points raised in [your above OT comment](https://github.com/ioccc-src/mkiocccentry/pull/135#issuecomment-1082478533).

BTW: Landon had a hand in the `chkconfig` code from System V Unix.  The `BSD` vs. `System V `"_so called debate_" was similar in style (and vitriol) to the more recent `systemd` vs `init` "_so called debate_" / "`flame wars`".

For the record, Landon much preferred  `BSD`  over  `System V ` Unix, but had to at times live within the limitations of `System V ` Unix.

Landon was is not a fan of some of the design decisions behind `systemd`: nevertheless he prefers to not get into `flame wars` over those design issues.  Instead Landon was a prolific filer of Red Hat Enterprise Linux bug reports (he had large number of systems with premium support contracts): which had an effect of helping Red Hat pay attention to critical issues. Such bug reports were carefully written to not offend as the `systemd` were known in the outside world has developing a _short fuse_: in part because of the `flame wars`.  Nevertheless it should have not taken 6-digit support contracts and very carefully written bug reports to encourage `systemd`  authors to fix basic functionality. 

**sigh**

xexyl commented:
> Interesting points raised in [your above OT comment](https://github.com/ioccc-src/mkiocccentry/pull/135#issuecomment-1082478533).
> 

Thank you!

> BTW: Landon had a hand in the `chkconfig` code from System V Unix. The `BSD` vs. `System V `"_so called debate_" was similar in style (and vitriol) to the more recent `systemd` vs `init` "_so called debate_" / "`flame wars`".
> 

Oh I totally hate `systemd` and I love how you have in the guidelines:

```
    We do not recommend submitting systemd source code to the IOCCC,
    if nothing else because that code is likely to exceed the source code
    size limit.  This isn't to say that another highly compact and obfuscated
    replacement of init would not be an interesting submission.
```

It's hilarious and so true. I really don't like `systemd`; the binary logging makes tools we had for years not useful (though it's possible to make it not use it one shouldn't have to do that) not to mention redundant logs that cause problems for users (I've seen reports of people thinking that their system had been compromised because of the wording of `systemd` logs without any context!).


> For the record, Landon much preferred `BSD` over `System V ` Unix, but had to at times live within the limitations of `System V ` Unix.
> 

It's been a long time since I've used BSD systems - unless you count the part of macOS I guess. I have fond memories of SunOS and later Solaris as well.


> Landon was is not a fan of some of the design decisions behind `systemd`: nevertheless he prefers to not get into `flame wars` over those design issues. Instead Landon was a prolific filer of Red Hat Enterprise Linux bug reports (he had large number of systems with premium support contracts): which had an effect of helping Red Hat pay attention to critical issues. Such bug reports were carefully written to not offend as the `systemd` were known in the outside world has developing a _short fuse_: in part because of the `flame wars`. Nevertheless it should have not taken 6-digit support contracts and very carefully written bug reports to encourage `systemd` authors to fix basic functionality.

Agreed. Yes their developers (I'm particularly looking at Lennart but I'm sure there are others) are short fused and it's why I have less sympathy for them. He's a hypocrite too when it comes to people having a go at him (like Linus himself did). 

It was a bad decision to go for `systemd` in the first place. I don't agree with Linus for saying that Lennart should be retroactively aborted but I can understand his frustration with him nonetheless.

Another one I hate is `dnf` instead of yum and particularly the logging in a simple file and format that makes it easy to see what packages were installed, updated, removed etc. But that's also OT so I'll end it at that.

> 
> **sigh**

I can relate to this! Thank you for the history btw.
